BuildMacOSUniversalBinary.py

```diff
@@ -162,6 +162,34 @@ def recursive_merge_binaries(src0, src1, dst):
        the source trees
     """
 
+    # Check that all files present in the folder are of the same type and that
+    # links link to the same relative location
+    for newpath0 in glob.glob(src0+"/*"):
+        filename = os.path.basename(newpath0)
+        newpath1 = os.path.join(src1, filename)
+        if not os.path.exists(newpath1):
+            continue
+
+        if os.path.islink(newpath0) and os.path.islink(newpath1):
+            p0 = os.path.realpath(newpath0)
+            p1 = os.path.realpath(newpath1)
+            if os.path.relpath(newpath0,src0) == os.path.relpath(newpath1,src1):
+                continue
+
+        if os.path.isdir(newpath0) and os.path.isdir(newpath1):
+            continue
+
+        # isfile() can be true for links so check that both are not links
+        # before checking if they are both files
+        if (not os.path.islink(newpath0)) and (not os.path.islink(newpath1)):
+            if os.path.isfile(newpath0) and os.path.isfile(newpath1):
+                continue
+
+        raise Exception(f"{newpath0} and {newpath1} cannot not be " +
```

Comment:
"cannot be" (there's an extra "not" in the message)

Reply:
Fixed